week3_assign/src/components/MainSectionWrapper.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import Prefer from "../Prefer/Prefer";
+import BtnSection from "./BtnSection";
+import ChooseButton from "./ChooseButton";
+import Title from "./Title";
+import Random from "../Prefer/Random/random";
+import Section from "./section";
+import Header from "./header";
+
+const MainSectionWrapper = () => {
+  const [currentStep, setCurrentStep] = useState("");
+  const resetStep = () => {
+    setCurrentStep("");
+  };
+  const Main = () => {
+    switch (currentStep) {
+      case "random":
+        return <Random></Random>;
+      case "prefer":
+        return <Prefer></Prefer>;
+      default:
+        return (
+          <Section>
+            <Title>원하는 추천방식</Title>
+            <BtnSection>
+              <ChooseButton
+                onClick={() => {
+                  setCurrentStep("prefer");
+                }}
+              >
+                <p>취향대로</p>
+              </ChooseButton>
+              <ChooseButton
+                onClick={() => {
+                  setCurrentStep("random");
+                }}
+              >
+                <p>랜덤으로</p>
+              </ChooseButton>
+            </BtnSection>
+          </Section>
+        );
+    }
+  };
+  return (
+    <>
+      <Header resetStep={resetStep}></Header>
+      <MainStyle>{Main()}</MainStyle>
```

Comment:
우리가 JSX 문법을 사용하기 때문에 element를 반환해주는건 맞는데, 우리는 그렇게 함으로써 `함수형 컴포넌트`를 생성해준거잖아용!? 따라서 일반적인 함수 호출 방식이 아닌, `<컴포넌트 />`를 사용해주면 됩니다!! 
족, 
```suggestion
      <Main />
```

📚 [JSX 공식문서 ](https://ko.legacy.reactjs.org/docs/introducing-jsx.html)
📚 [Components 공식 문서](https://ko.legacy.reactjs.org/docs/components-and-props.html)